### PR TITLE
RAG DEMO Using Intel AMX

### PR DESCRIPTION
**Pull request by joshua-segovia-intel**
Initial RAG DEMO utilizing gradio front end with GPT4ALL GGUF models to showcase a RAG based LLM

### CODE REVIEW

File: recipes/ai-rag-amx-ubuntu/terraform.tfstate [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
```
Comment (lucasmelogithub) [inline recipes/ai-rag-amx-ubuntu/terraform.tfstate]: We need to remove this file from the commit and add *.tfstate to the .gitignore file

**Comment by lucasmelogithub** [comment]
@joshua-segovia-intel @wsfowler Please remove the tfstate file from the commit and we can merge this initial version. 
And we should update the README.md soon after.

File: recipes/ai-rag-amx-ubuntu/README.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+https://gpt4all.io/index.html
+
+## Usage
+
+[**AWS - Intel® Optimized Cloud Modules for HashiCorp Terraform example**](https://github.com/intel/terraform-intel-aws-vm/tree/fast-rag/examples/gen-ai-rag-demo) 
+
+## Running the Demo
+
+1. SSH into newly created GCP VM and run `source /usr/local/tmp/optimized-cloud-recipes/recipes/ai-rag-ubuntu/gradio_rag.py`
```
Comment (wsfowler) [inline recipes/ai-rag-amx-ubuntu/README.md]: Should this be AWS or just generic?

**Comment by chrisahsiong23** [comment]
Cleaned up the repo with your suggested modifications. Please let us know if there are any more changes need to be made!

File: recipes/ai-rag-amx-ubuntu/README.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+# Intel® Corporation Copyright
+
+**Contributors: William Fowler, Joshua Segovia, Chris Ah-Siong**
+
+# Intel® Gen AI RAG DEMO using Xeon AMX
+
+## Description 
+
+The purpose of this demo aims to demonstrate the powerful capabilities of retrieval augmented generation based on large language models deployed on an Intel Xeon AWS instance (M7i.8xlarge), where we highlight performance optimizations leveraging Intel instructions to deliver increased accuracy and speed of inference.
```
Comment (lucasmelogithub) [inline recipes/ai-rag-amx-ubuntu/README.md]: Could we remove all comments around AWS and the Instance  `Intel Xeon AWS instance (M7i.8xlarge).`
Use `4th or 5th Gen Intel Xeon instances with Intel AMX` instead?

File: recipes/ai-rag-amx-ubuntu/README.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+https://gpt4all.io/index.html
+
+## Usage
+
+[**AWS - Intel® Optimized Cloud Modules for HashiCorp Terraform example**](https://github.com/intel/terraform-intel-aws-vm/tree/fast-rag/examples/gen-ai-rag-demo) 
+
+## Running the Demo
+
+1. SSH into newly created AWS VM and run `source /usr/local/tmp/optimized-cloud-recipes/recipes/ai-rag-ubuntu/gradio_rag.py`
```
Comment (lucasmelogithub) [inline recipes/ai-rag-amx-ubuntu/README.md]: Remove AWS

File: recipes/ai-rag-amx-ubuntu/README.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+In particular, the non-intel models utilize AVX instructions, enhancing computational efficiency for real time information retrieval. The intel trained neural chat model further benefits from AMX, providing additional performance improvements for handling complex matrix operations. These optimizations enable users to perform inference without the need for specialized hardware and can deliver high-quality results efficiently on Intel Xeon CPUs. 
+
+Additionally, we implement RAG with LangChain, which retrieves relevant context or passages from the datasets (ranging from robot maintenance to minor healthcare consultations). The chatbot then utilizes the LLM to generate contextually relevant responses based on both the retrieved context, and the user query. 
+
+https://gpt4all.io/index.html
+
+## Usage
+
+[**AWS - Intel® Optimized Cloud Modules for HashiCorp Terraform example**](https://github.com/intel/terraform-intel-aws-vm/tree/fast-rag/examples/gen-ai-rag-demo)
```
Comment (lucasmelogithub) [inline recipes/ai-rag-amx-ubuntu/README.md]: This should point to `main` branch instead of `fast-rag`

**Pull request closed without merging**